### PR TITLE
Relevant events

### PR DESCRIPTION
**Pull request by ptormene**
Now the OQ-Engine return only the relevant events, so the list of event ids might likely have missing values, so we need to help the user identifying the closest event id with respect to the initially chosen one.
I am also fixing some minor aspects due to recent changes in the engine and in the numpy api.

### CODE REVIEW

File: svir/dialogs/load_gmf_data_as_layer_dialog.py [hunk 1]
```diff
@@ -69,11 +69,15 @@ def __init__(self, drive_engine_dlg, iface, viewer_dock, session, hostname,
             self.on_extract_error)
         QgsApplication.taskManager().addTask(self.extract_npz_task)
 
+    def get_closest_element(self, element, elements):
+        return min(elements, key=lambda x: abs(x - element))
+
```
Comment (micheles) [inline svir/dialogs/load_gmf_data_as_layer_dialog.py:74]: The idea is good, but you need to use numpy for performance:
```
return elements[numpy.abs(elements - element).argmin()]        
```

File: svir/dialogs/load_gmf_data_as_layer_dialog.py [hunk 2]
```diff
@@ -86,12 +90,42 @@ def get_eid(self, events_npz):
...
+                self.eid, ok = QInputDialog.getInt(
+                    self.drive_engine_dlg,
+                    "Select an event ID", "Range (%s - %s)" % (
+                        events[0]['id'], events[-1]['id']),
+                    self.eid, events[0]['id'], events[-1]['id'])
+                if not ok:
+                    self.reject()
+                    return
         if not ok:
             self.reject()
```
Comment (micheles) [inline svir/dialogs/load_gmf_data_as_layer_dialog.py:130]: I do not like the duplication of logic

**Comment by micheles** [comment]
The tests are red

File: svir/dialogs/load_gmf_data_as_layer_dialog.py (hunk 1 above)
Comment (ptormene) [reply svir/dialogs/load_gmf_data_as_layer_dialog.py:74]: Thanks. I updated it as suggested.

File: svir/dialogs/load_gmf_data_as_layer_dialog.py (hunk 2 above)
Comment (ptormene) [reply svir/dialogs/load_gmf_data_as_layer_dialog.py:115]: I've removed the duplication of logic

**Comment by micheles** [comment]
LGTM!

**Pull request merged**